### PR TITLE
Add EmberComponent type

### LINKED ISSUE #100: Add EmberComponent type
Get implementation code from [here](https://github.com/dogma-io/ember-dogma-core/blob/master/addon/helpers/is-component.js).

### PR DESCRIPTION
**Pull request by sandersky**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

Resolves #100 

# CHANGELOG

* **Added** `EmberComponent` prop-type for properties that come from the `{{component}}` helper.

### CODE REVIEW

File: addon/utils/validators/ember-component.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import Ember from 'ember'
+const {typeOf} = Ember
+
+import logger from '../logger'
+
+export default function (ctx, name, value, def, logErrors, throwErrors) {
+  const isObject = typeOf(value) === 'object'
+
+  const valid = isObject && Object.keys(value).some((key) => {
+    return key.indexOf('COMPONENT_CELL') === 0
```
Comment (sglanzer) [inline addon/utils/validators/ember-component.js]: Can we flag this in some was as "internal", just in case things break we'll know where to look first

**Comment by sglanzer** [comment]
Approved

File: addon/utils/validators/ember-component.js (hunk 1 above)
Comment (sandersky) [inline addon/utils/validators/ember-component.js]: Do you simply adding a comment like below?
```
NOTE: this is mimicking internal API and thus could break without warning.
```

**Comment by sandersky** [comment]
:+1:

**Pull request merged**